marlowe-playground-client/src/Simulation/State.purs

```diff
@@ -226,9 +252,18 @@ updatePossibleActions oldState =
 
     usefulActions = mapMaybe removeUseless actions
 
-    actionInputs = Map.fromFoldable $ map (actionToActionInput nextState) usefulActions
+    currentSlot = oldState ^. _slot
+
+    slot = fromMaybe (add one currentSlot) (nextSignificantSlot oldState)
+
+    actionInputs = Map.fromFoldable $ (map (actionToActionInput nextState) usefulActions)
+
+    moveTo = if oldState ^. (_contract <<< to (eq (Just Close))) then Nothing else Just $ MoveToSlot slot
```

Comment:
I'm not sure about lifecycle of updatePossibleActions here. In UI I see Other Actions on top of possible ones sometimes. I suspect it happens when previous state is empty. 

Does action composer differentiate between transition from one contract to another and reducing existing contract?

Reply:
Hmm, that's an interesting ordering question, I can look into it and make sure _Other Actions_ is always at the bottom

Reply:
fixed